api/nodes/serializers.py

```diff
@@ -736,14 +751,7 @@ def update(self, node, validated_data):
                 node.save()
             if 'subjects' in validated_data:
                 subjects = validated_data.pop('subjects', None)
-                try:
-                    node.set_subjects(subjects, auth)
-                except PermissionsError as e:
-                    raise exceptions.PermissionDenied(detail=str(e))
-                except ValueError as e:
-                    raise exceptions.ValidationError(detail=str(e))
-                except NodeStateError as e:
-                    raise exceptions.ValidationError(detail=str(e))
+                self.update_subjects(node, subjects, auth)
```

Comment:
Same as above, `update_subjects` method is now covering the error handling plus determining which method to use when populating subjects (subjects may either be coming in as attributes or relationships with different formatting.